pkg/record/handler/save.go

```diff
@@ -361,13 +372,62 @@ func (f RecordFetcher) FetchOrCreateRecord(recordID record.ID, authInfo *authinf
 // 3. Clean up some transport only data (sequence for example) away from record
 // 4. Populate meta data and save the record (like updated_at/by)
 // 5. Execute after save hooks with original record and new record
-func RecordSaveHandler(req *RecordModifyRequest, resp *RecordModifyResponse) skyerr.Error {
+// nolint: gocyclo
+func RecordSaveHandler(req *RecordModifyRequest, resp *RecordModifyResponse) (err error) {
 	records := req.RecordsToSave
 
-	fetcher := NewRecordFetcher(req.Context, req.RecordStore, req.WithMasterKey)
+	// Open trasaction for whole record save operation
+	if req.Atomic {
```

Comment:
Is it possible to move all atomic related code to the `RecordSaveHandler.Handle`? Keep this RecordSaveHandler transaction unaware may be better separate of concern.

This `RecordSaveHandler` already complicated enough.